src/app/controllers/location/create.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+/**
+ * @ngdoc function
+ * @name flightNodeApp.controller.location:LocationCreateController
+ * @description
+ * # LocationCreateController
+ * Controller for the create work type page.
+ */
+angular.module('flightNodeApp')
+	.controller('LocationCreateController',
+		['$scope', '$http', '$log', '$location', 'messenger', 'oauthRequest',
+			function ($scope, $http, $log, $location, messenger, oauthRequest) {
+
+				$scope.loading = true;
+
+				$scope.cancel = function () {
+					$location.path('/locations');
+				}
+
+				$scope.submit = function () {
+					$scope.loading = true;
+
+					oauthRequest.post('http://localhost:50323/api/v1/locations', $scope.location)
+						.then(function success(response) {
+
+							messenger.showSuccessMessage($scope, 'Saved');
+							$scope.loading = false;
+
+						}, function error(response) {
+
+							switch (response.status) {
```

Comment:
This full switch statement is used over and over again, in many files now. I need to extract it into a utility function, but at the moment I'm more interested in usable functionality and refactoring soon.